allennlp/semparse/domain_languages/domain_language.py

```diff
@@ -12,6 +13,41 @@
 
 logger = logging.getLogger(__name__)
 
+
+# We rely heavily on the typing module and its type annotations for our grammar induction code.
+# Unfortunately, the behavior of the typing module changed somewhat substantially between python
+# 3.6 and 3.7, so we need to do some gymnastics to get some of our checks to work with both.
+# That's what these three methods are about.
+
+def is_callable(type_: Type) -> bool:
+    if sys.version_info[1] == 6:  # python 3.6
```

Comment:
I feel like these checks would be slightly more readable if you wrote them as

```
if sys.version_info < (3, 7)
```
what do you think?

Reply:
Does checking a tuple work like that?  I didn't realize you could do that.  Fancy.  Assuming it works, I'll do that instead.

Reply:
yeah, tuples compare lexicographically, so e.g. (3, 6, 8) < (3, 7)

Reply:
Thanks for the suggestion, yeah, I like that a lot better.